src/main/java/app/controllers/StudentController.java

```diff
@@ -17,136 +23,272 @@
+            }
+
+        } catch (SQLException e) {
+            System.out.println("Erro ao buscar o ID do critério: " + e.getMessage());
+        } finally {
+            try {
+                if (resultSet != null) resultSet.close();
+                if (statement != null) statement.close();
+                if (connection != null) connection.close();
+            } catch (SQLException e) {
+                System.out.println("Erro ao fechar recursos: " + e.getMessage());
+            }
+        }
+        return idCriterio;
+    }
+
+    public Integer obterIdDaSprint() {
+        String sprintSelecionada = choiceBoxMudarSprint.getValue();
+        Integer idSprint = sprintIdMap.get(sprintSelecionada);
+
+        return idSprint;
+
+    }
+
+    public Integer obterIdDoAvaliado(String nomeAluno) {
+        Integer idAluno = null;
+        try {
+            connection = DatabaseConnection.getConnection(true);
+            String sql = "SELECT id FROM usuario WHERE nome = ?";
+            statement = connection.prepareStatement(sql);
+            statement.setString(1, nomeAluno);
+            resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                idAluno = resultSet.getInt("id");
+            }
+
+        } catch (SQLException e) {
+            System.out.println("Erro ao buscar o ID do aluno: " + e.getMessage());
+        } finally {
+            try {
+                if (resultSet != null) resultSet.close();
+                if (statement != null) statement.close();
+                if (connection != null) connection.close();
+            } catch (SQLException e) {
+                System.out.println("Erro ao fechar recursos: " + e.getMessage());
+            }
+        }
+        return idAluno;
+    }
+
     @FXML
-    public void goToLoginScreen(ActionEvent event) {
-        Utils.setScreen(event, "loginScreen");
+    public void ConfirmarNotas() {
+        Connection connection = null;
+        PreparedStatement statementNota = null;
+
+        for (AvaliacaoModel aluno : tableView.getItems()) {
+            for (TableColumn<AvaliacaoModel, ?> column : tableView.getColumns()) {
+                if (!column.getText().equals("Aluno")) {
+
+                    Integer notaValor = (Integer) column.getCellData(aluno);
+
+                    if (notaValor != null) {
+                        String nomeCriterio = column.getText();
+                        Integer idCriterio = obterIdDoCriterio(nomeCriterio);
+
+                        String nomeAluno = aluno.getNome();
+                        Integer idAvaliado = obterIdDoAvaliado(nomeAluno);
+
+                        Integer idSprint = obterIdDaSprint();
+
+                        try {
+                            connection = DatabaseConnection.getConnection(true);
+                            String sqlNota = "INSERT INTO nota (valor, avaliador, avaliado, criterio, periodo, sprint) VALUES (?, ?, ?, ?, ?, ?)";
```

Comment:
voce está dando insert. Está funcionando, mas seria bom colocar uma validação pra nao executar esse metodo caso ja tenha valores nessa sprint, e avisar o usuario disso.
Exemplo: Meu banco já está populado, então os valores que eu coloco nessa tela não estão sendo sobrescritos, porque vc ta usando insert e nao update. E realmente nao deve sobrescrever e nem gerar registros novos. MAS, seria bom voce fazer uma query antes `SELECT valor FROM nota WHERE periodo = ? AND sprint = ?`, e caso retorne um resultSet vazio, voce faz o seu INSERT, se voltar com informações, usa um Utils.setAlert e avisa o usuário que ele já avaliou nesse periodo e sprint